server/budgettracker.data/Repositories/BudgetRepository.cs

```diff
@@ -30,12 +31,41 @@ public async Task<Budget> CreateBudget(Budget budget)
                 await _dbContext.Budgets.AddAsync(newBudget);
                 int recordSaved = await _dbContext.SaveChangesAsync();
 
-                if(recordSaved != 1) 
+                if(recordSaved < 1) 
                 {
                     throw new RepositoryException("Created " + recordSaved + " budget(s) when only 1 should have been created");
                 }
             }
             return _budgetConverter.ToBusinessModel(newBudget);  
         }
+
+        public async Task<Budget> UpdateBudget(Budget budget)
+        {
+            BudgetModel oldBudget;
+            using (_dbContext)
```

Comment:
Don't use using here. The using statement basically destroys the context afterwards which means you can't use this repository any more. We don't want that.

Reply:
We the repository is created every call with the service.AddTransiente correct so it would get recreated every time anyway?

Reply:
But within 1 call, we may use it multiple times. For example, the delete function might delete the object, send the sql query, then query the DB to make sure it was deleted. If you wrap the delete call in a using, you can't re-query to check if it worked.